src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -84,42 +42,25 @@ public virtual int WriteTimeout
             set => throw new InvalidOperationException(SR.InvalidOperation_TimeoutsNotSupported);
         }
 
-        public Task CopyToAsync(Stream destination)
-        {
-            int bufferSize = GetCopyBufferSize();
-
-            return CopyToAsync(destination, bufferSize);
-        }
-
-        public Task CopyToAsync(Stream destination, int bufferSize)
-        {
-            return CopyToAsync(destination, bufferSize, CancellationToken.None);
-        }
-
-        public Task CopyToAsync(Stream destination, CancellationToken cancellationToken)
-        {
-            int bufferSize = GetCopyBufferSize();
-
-            return CopyToAsync(destination, bufferSize, cancellationToken);
-        }
+        public void CopyTo(Stream destination) => CopyTo(destination, GetCopyBufferSize());
 
-        public virtual Task CopyToAsync(Stream destination, int bufferSize, CancellationToken cancellationToken)
+        public virtual void CopyTo(Stream destination, int bufferSize)
         {
-            StreamHelpers.ValidateCopyToArgs(this, destination, bufferSize);
-
-            return CopyToAsyncInternal(destination, bufferSize, cancellationToken);
-        }
+            ValidateCopyToArguments(destination, bufferSize);
+            if (!CanRead)
+            {
+                throw CanWrite ? (Exception)
+                    new NotSupportedException(SR.NotSupported_UnreadableStream) :
+                    new ObjectDisposedException(GetType().Name, SR.ObjectDisposed_StreamClosed);
+            }
 
-        private async Task CopyToAsyncInternal(Stream destination, int bufferSize, CancellationToken cancellationToken)
-        {
             byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
             try
             {
-                while (true)
+                int read;
```

Comment:
nit: can we call this "bytesRead"? That's the standard var name for the result of Read

Reply:
(same below in async version)

Reply:
Sure.  I mapped the async one to use the same name as the sync one, but I can reverse the change :)